### DISCUSSION TITLE
resampled qsm could’t match with qsm_segmentations

### DISCUSSION THREAD

**Discussion by lhldrs** [QSMxT/QSMxT#92] [General]
HI!
   I conducted  39 qsm images by QSMxT(run_2_qsm.py and run_3_segment.py run without bug), and some results were resampled. Run_5_analysis.py run with a bug, which seemed that resampled qsm images didn’t match with qsm_segmentations. Resampled qsm images also didn’t match with qsm_segmentations in ITK-SNAP.

**Comment by astewartau** [comment]
Ah, thanks for raising this problem! A quick workaround for this will be to run the `run_2_qsm.py` pipeline with the parameter `--obliquity_threshold 999`. This will prevent the resampling from occurring in the QSM pipeline at all. 

A better solution may be to modify the `run_5_analysis.py` file so that it resamples non-matching images to ensure that they do match. I will add this as an issue to work on. https://github.com/QSMxT/QSMxT/issues/93

**Reply by lhldrs** [reply]
Thank you very much!!!

**Reply by lhldrs** [reply]
When I tried to the run_2_qsm.py pipeline with the parameter --obliquity_threshold 999, a new error occurred. The code and the bug are as follows: 
python3 /opt/QSMxT/run_2_qsm.py   01_bidstest 02_qsm_output --obliquity_threshold 999;
run_2_qsm.py: error: unrecognized arguments: --obliquity_threshold.

**Reply by astewartau** [reply]
Hi @lhldrs, thanks for letting me know! The `--obliquity_threshold` parameter was actually only available in the current master branch, rather than in the latest container build v1.2.0 as I thought it was. You can pull the master branch code and run the `run_2_qsm.py` script from inside your QSMxT container to run it with this setting. 

Alternatively, you can use the next version, v1.3.0, which I will be releasing shortly. This will have the `--obliquity_threshold` setting, as well as a much more user-friendly interface for selecting a pipeline and adjusting pipeline settings. :)

**Reply by astewartau** [reply]
Hi again, @lhldrs! The new QSMxT version v1.3.2 (.2 because there were a few bugs quickly identified) has been released. This version should interactively assist you to skip the axial resampling step in the phase processing settings. :)

As long as you set the obliquity threshold to `-1` or a large number such as `999`, the resampling step will be skipped, and any segmentations should align correctly with your results. 

Let me know if you have any further issues. I will aim for a future version to also include the resampling step in the analysis pipeline so that they correctly match and do not require you to skip it in future.
